src/vs/workbench/api/browser/mainThreadWebview.ts

```diff
@@ -589,51 +595,125 @@ class MainThreadCustomEditorModel extends Disposable implements ICustomEditorMod
 	readonly onDidChangeContent: Event<void> = this._onDidChangeContent.event;
 
 	//#endregion
-
 	public get viewType() {
 		return this._viewType;
 	}
 
-	public setDirty(dirty: boolean): void {
+	public pushEdit(editId: number) {
+		if (!this._editable) {
+			throw new Error('Document is not editable');
+		}
+
+		this.change(() => {
+			this.spliceEdits(editId);
+			this._currentEditIndex = this._edits.length - 1;
+		});
+
+		this._undoService.pushElement({
+			type: UndoRedoElementType.Resource,
+			resource: this.resource,
+			label: 'Undo',
```

Comment:
This label will be used in the context "Undo {0}", so it would be better if it was called "Edit", because now it might be presented as "Undo Undo"

Reply:
Thanks for clarifying. I'll also investigate getting the `label` from extensions